docs/advanced/bootloader.md

```diff
@@ -0,0 +1,89 @@
+# Bootloader
+
+This guide explains how to create a bootloader, how to create a main program to go with the bootloader and how you can use this bootloader to perform software updates.
```

Comment:
It's a weird switch from "how to create" (which is sort of vague) to "how you can use this" (which is very specific)